rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -522,6 +551,235 @@ void save_callsFederationSupportSave() throws IOException {
+            when(rskTx.getHash()).thenReturn(rskTxHash);
+
+            BridgeEventLogger bridgeEventLogger = new BridgeEventLoggerImpl(bridgeMainNetConstants, activationsAfterForks, logs, signatureCache);
+
+            bridgeSupport = bridgeSupportBuilder
+                .withBridgeConstants(bridgeMainNetConstants)
+                .withEventLogger(bridgeEventLogger)
+                .withProvider(bridgeStorageProvider)
+                .withActivations(activationsAfterForks)
+                .withFederationSupport(federationSupport)
+                .withFeePerKbSupport(feePerKbSupport)
+                .withExecutionBlock(rskExecutionBlock)
+                .build();
+        }
+
+        @Test
+        void processSvpFundTransactionWithoutSignatures_whenProposedFederationDoesNotExist_throwsIllegalStateException() {
+            // arrange
+            when(federationSupport.getProposedFederation()).thenReturn(Optional.empty());
+
+            // act & assert
+            IllegalStateException exception = assertThrows(IllegalStateException.class,
+                () -> bridgeSupport.processSvpFundTransactionUnsigned(rskTx));
+
+            String expectedMessage = "Proposed federation should be present when processing SVP fund transaction.";
+            assertEquals(expectedMessage, exception.getMessage());
+        }
+
+        @Test
+        void processSvpFundTransactionWithoutSignatures_createsExpectedTransactionAndSavesTheHashInStorageEntryAndPerformsPegoutActions() throws Exception {
+            // act
+            bridgeSupport.processSvpFundTransactionUnsigned(rskTx);
+            bridgeStorageProvider.save(); // to save the tx sig hash
+
+            // assert
+            assertSvpFundTxHashUnsignedWasSavedInStorage();
+            assertSvpReleaseWasSettled();
+            assertSvpFundTransactionHasExpectedInputsAndOutputs();
+        }
+
+        private void assertSvpFundTxHashUnsignedWasSavedInStorage() {
+            Optional<Sha256Hash> svpFundTransactionHashUnsignedOpt = bridgeStorageProvider.getSvpFundTxHashUnsigned();
+            assertTrue(svpFundTransactionHashUnsignedOpt.isPresent());
+
+            svpFundTransactionHashUnsigned = svpFundTransactionHashUnsignedOpt.get();
+        }
+
+        private void assertSvpReleaseWasSettled() throws IOException {
+            PegoutsWaitingForConfirmations pegoutsWaitingForConfirmations = bridgeStorageProvider.getPegoutsWaitingForConfirmations();
+            assertPegoutWasAddedToPegoutsWaitingForConfirmations(pegoutsWaitingForConfirmations, svpFundTransactionHashUnsigned, rskTxHash);
+            getSvpFundTransactionFromPegoutsMap(pegoutsWaitingForConfirmations);
+
+            assertPegoutTxSigHashWasSaved(svpFundTransactionUnsigned);
+            assertLogReleaseRequested(rskTxHash, svpFundTransactionHashUnsigned, bridgeMainNetConstants.getSpendableValueFromProposedFederation());
+            assertLogPegoutTransactionCreated(svpFundTransactionUnsigned);
```

Comment:
ideally we should use an `assertReleaseRequestWasSettled` that call all these methods, like in prod code. But in this case we have to "recover" the transaction from the pegouts map, first asserting it is there using its hash.